src/pins/pin.rs

```diff
@@ -427,3 +525,24 @@ where
         }
     }
 }
+
+impl GenericPin {
+    /// Creates a new `GenericPin`
+    pub fn new(port: usize, id: u8) -> Self {
```

Comment:
Still public, which breaks pin uniqueness!

```suggestion
    pub(super) fn new(port: usize, id: u8) -> Self {
```

I didn't carefully check if `pub(super)` is enough, maybe we need `pub(crate)` here.